torch/linalg/__init__.py

```diff
@@ -378,6 +378,73 @@
             [1, 2, 2, 2]])
 """)
 
+vector_norm = _add_docstr(_linalg.linalg_vector_norm, r"""
+linalg.vector_norm(input, ord=None, keepdim=False, *, out=None, dtype=None) -> Tensor
+
+Returns the vector norm of a given tensor. 
+
+Args:
+    input (Tensor): The input tensor. If the input has more than one dimension,
+        it is interpretted as a tensor flattened to one dimension.  The input
+        data type must be either a floating point or complex type. For complex
+        inputs, the norm is calculated on of the absolute values of each
+        element. If the input is complex and neither :attr:`dtype` nor
+        :attr:`out` is specified, the result's data type will be the
+        corresponding floating point type (e.g. float if :attr:`input` is
+        complex float).
+
+    ord (int, float, inf, -inf, optional): The order of norm.
+        inf refers to :attr:`float('inf')`, numpy's :attr:`inf` object, or any
+        equivalent object. The following norms can be calculated:
+
+        =====  ==========================
+        ord    norm
+        =====  ==========================
+        None   2-norm
+        inf    max(abs(x))
+        -inf   min(abs(x))
+        0      sum(x != 0)
+        1      as below
+        -1     as below
+        2      as below
+        -2     as below
+        other  sum(abs(x)**ord)**(1./ord)
+        =====  ==========================
+
+        Default: ``None``
+
+    keepdim (bool, optional): If set to True, the reduced dimensions are retained
+        in the result as dimensions with size one. Default: ``False``
+
+Keyword args:
+
+    out (Tensor, optional): The output tensor. Ignored if ``None``. Default: ``None``
+
+    dtype (:class:`torch.dtype`, optional): If specified, the input tensor is cast to
+        :attr:`dtype` before performing the operation, and the returned tensor's type
+        will be :attr:`dtype`. If this argument is used in conjunction with the
```

Comment:
This limitation is OK for now but in the future we should support safe casting.

cc @IvanYashchuk who just fixed a swathe of linalg functions. Depending on when those PRs land it may be convenient to update this op's behavior in this PR, too.

Reply:
Just for my own reference, I believe this is the PR you're talking about: #51560